flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogFunctionImpl.java

```diff
@@ -30,10 +30,17 @@
  */
 public class CatalogFunctionImpl implements CatalogFunction {
 	private final String className; // Fully qualified class name of the function
+	private final FunctionLanguage functionLanguage;
 
 	public CatalogFunctionImpl(String className) {
+		this(className, FunctionLanguage.JAVA);
+	}
+
+	public CatalogFunctionImpl(String className, FunctionLanguage functionLanguage) {
 		checkArgument(!StringUtils.isNullOrWhitespaceOnly(className), "className cannot be null or empty");
+		checkArgument(functionLanguage != null, "functionLanguage cannot be null");
 		this.className = className;
+		this.functionLanguage = functionLanguage;
```

Comment:
this can be `this.functionLanguage=checkNotNull(functionLanguage, "functionLanguage cannot be null")`

Reply:
Done